lib/ui/window/pointer_data_packet_converter.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_WINDOW_POINTER_DATA_PACKET_CONVERTER_H_
+#define FLUTTER_LIB_UI_WINDOW_POINTER_DATA_PACKET_CONVERTER_H_
+
+#include <string.h>
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/window/pointer_data_packet.h"
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The current information about a pointer. This struct is used by
+/// PointerDataPacketConverter to fill in necesarry information for raw pointer
+/// packet sent from embedding.
+///
+struct PointerState {
+  int64_t pointer;
+  bool isDown;
+  double physical_x;
+  double physical_y;
+};
+
+//------------------------------------------------------------------------------
+/// Converter to expend the raw pointer data packet from the platforms.
+///
+/// All pointer data packets that are sent from all the platforms must go
+/// through this converter before sending them to framework.
+///
+class PointerDataPacketConverter {
+ public:
+  PointerDataPacketConverter();
+  ~PointerDataPacketConverter();
+
+  //----------------------------------------------------------------------------
+  /// @brief      Expands pointer data packet into a form that framework
+  ///             understands. The raw pointer data packet from embedding does
+  ///             not have sufficient information and may contain illegal
```

Comment:
Please also elaborate on an example that this corrects. Also, document the transitions that this class considers legal.

Reply:
I included the example in class doc string. I feel a bit redundant to include again in here